compiler/passes/createTaskFunctions.cpp

```diff
@@ -370,16 +370,21 @@ static void addReduceIntentSupport(FnSymbol* fn, CallExpr* call,
   Expr* tailAnchor = findTailInsertionPoint(call, isCoforall);
   // Doing insertAfter() calls in reverse order.
   // Can't insertBefore() on tailAnchor->next - that can be NULL.
-  tailAnchor->insertAfter("chpl__delete(%S)",
-                         globalOp);
-  tailAnchor->insertAfter("'='(%S, generate(%S,%S))",
-                         origSym, gMethodToken, globalOp);
+  // origSym = <- globalOp.generate() (via genTemp); delete globalOp;
+  VarSymbol* genTemp = newTemp("genTemp");
+  genTemp->addFlag(FLAG_INSERT_AUTO_DESTROY);
+  tailAnchor->insertAfter(new CallExpr("chpl__delete", globalOp));
+  tailAnchor->insertAfter(new CallExpr("=", origSym, genTemp));
+  tailAnchor->insertAfter("'move'(%S,generate(%S,%S))",
+                          genTemp, gMethodToken, globalOp);
+  tailAnchor->insertAfter(new DefExpr(genTemp));
```

Comment:
Inserting things in reverse order is in my opinion unnecessarily confusing.
Here is how I would write this:
``` c++
  CallExpr* del = new CallExpr("chpl__delete", globalOp);
  tailAnchor->insertAfter(del);
  del->insertBefore(new DefExpr(genTemp));
  del->insertBefore(new CallExpr(PRIM_MOVE, genTemp, new CallExpr("generate", gMethodToken, globalOp));
  del->insertBefore(new CallExpr("=", origSym, genTemp));
```

but using a block or a PRIM_NOOP would also be reasonable alternatives.

If you took my multi-arg insertAfter suggestion, it would look like this
``` c++
  tailAnchor->insertAfter( 
    new DefExpr(genTemp),
    new CallExpr(PRIM_MOVE, genTemp, new CallExpr("generate", gMethodToken, globalOp)),
    new CallExpr("=", origSym, genTemp)),
    new CallExpr("chpl__delete", globalOp));
```